app/features/recent-list/reducer.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+
+import { CONFERENCE_ENDED, CONFERENCE_JOINED } from '../conference';
+
+import type { RecentListItem } from './types';
+
+type State = {
+    recentList: Array<RecentListItem>;
+};
+
+const DEFAULT_STATE = {
+    recentList: []
+};
+
+/**
+ * Reduces redux actions for features/settings.
+ *
+ * @param {State} state - Current reduced redux state.
+ * @param {Object} action - Action which was dispatched.
+ * @returns {State} - Updated reduced redux state.
+ */
+export default (state: State = DEFAULT_STATE, action: Object) => {
+    switch (action.type) {
+    case CONFERENCE_ENDED:
+        return {
+            ...state,
+            recentList: _updateEndtimeOfLastConference(state.recentList)
+        };
+
+    case CONFERENCE_JOINED:
+        return {
+            ...state,
+            recentList: _insertConference(state.recentList, action.conference)
+        };
+
+    default:
+        return state;
+    }
+};
+
+/**
+ * Insert Conference details in the recent list array.
+ *
+ * @param {Array<RecentListItem>} recentList - Previous recent list array.
+ * @param {RecentListItem} newConference - Conference that has to be added
+ * to recent list.
+ * @returns {Array<RecentListItem>} - Updated recent list array.
+ */
+function _insertConference(
+        recentList: Array<RecentListItem>,
+        newConference: RecentListItem
+) {
+
+    // Add start time to conference.
+    newConference.startTime = Date.now();
+
+    // Remove same conference.
+    // $FlowFixMe
```

Comment:
what does flow complain about?

Reply:
Huh... It no longer does... Thanks for catching that :smile: